src/seccompiler/src/types.rs

```diff
@@ -0,0 +1,177 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+use std::ffi::CString;
+use std::str::FromStr;
+
+use serde::*;
+
+// use libseccomp::{ScmpAction, ScmpArch, ScmpCompareOp};
+use crate::bindings::*;
+
+/// Comparison to perform when matching a condition.
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum SeccompCmpOp {
+    Eq,
+    Ge,
+    Gt,
+    Le,
+    Lt,
+    MaskedEq(u64),
+    Ne,
+}
+
+/// Condition that syscall must match in order to satisfy a rule.
+#[derive(Debug, Deserialize)]
+pub struct SeccompCondition {
+    pub index: u8,
+    pub op: SeccompCmpOp,
+    pub val: u64,
+}
+
+impl SeccompCondition {
+    pub fn to_scmp_type(&self, arch: &TargetArch, syscall: i32) -> scmp_arg_cmp {
+        // For `ioctls` we need to mask upper bits as musl
+        // sets them to 1, but libseccomp explicitly checks that they are 0.
+        // with 0x00000000FFFFFFFF mask upper bits are always 0.
+        let ioctl = match arch {
+            TargetArch::X86_64 => 16,
+            TargetArch::Aarch64 => 29,
```

Comment:
if we passed syscall as a string, then we can resolve it here and not have to hardcode these values.